### PR TITLE
Add additional fine fee types

### PR DESCRIPTION
**Pull request by adamshire123**
### What does this PR do?
This PR adds additional fine fee type mappings from the Alma export file, to the target CSV file description field

### Helpful background context

During stakeholder testing, it was requested to add these additional fine fee types, which 
are not currently in use, but my be used in the future.

### Includes new or updated dependencies?

YES

Notably, Moto was updated to v5 requiring an update to conftest.py

### What are the relevant tickets?

* https://mitlibraries.atlassian.net/browse/ENSY-182
### Developer

- [x] All new ENV is documented in README (or there is none)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: lambdas/bursar_transfer.py [hunk 1]
```diff
@@ -115,10 +115,19 @@ def generate_description(fine_fee_type: str, barcode: str) -> str:
     that an unexpected change has been made in the Alma bursar
     integration config.
     """
-    if "overdue" in fine_fee_type.lower():
-        mapped_type = "Library overdue"
-    elif "lost" in fine_fee_type.lower():
+    if fine_fee_type == "DAMAGEDITEMFINE":
```
Comment (ehanson8) [inline lambdas/bursar_transfer.py:118]: Optional: a pattern I've used to reduce the `elif`s is creating a `dict` and iterating through the items like this:

```
aardvark_location_fields = {
            "dcat_bbox": "Bounding Box",
            "locn_geometry": "Geometry",
        }
for aardvark_location_field, kind_value in aardvark_location_fields.items():
```

But this does work so again, totally optional

**Review by ehanson8 — APPROVED** [review]
Looks good, appreciate the extra detail in the commit message explaining the changes

File: lambdas/bursar_transfer.py (hunk 1 above)
Comment (ghukill) [reply lambdas/bursar_transfer.py:118]: I'd second that a single dictionary that provides a mapping from the uppercase key to the human friendly form could be a nice fit here.  But optional and non-blocking as well; I'm also partial to crystal cleaer if/elif structures like this.

**Review by ghukill — APPROVED** [review]
Looks good!

File: lambdas/bursar_transfer.py [hunk 2]
```diff
@@ -107,31 +107,18 @@ def generate_description(fine_fee_type: str, barcode: str) -> str:
...
+        "DAMAGEDITEMFINE": "Library damaged",
+        "LOSTITEMPROCESSFEE": "Library lost",
+        "LOSTITEMREPLACEMENTFEE": "Library repl",
+        "OVERDUEFINE": "Library overdue",
+        "OTHER": "Library other",
+        "RECALLEDOVERDUEFINE": "Library recalled",
+    }
 
-    return f"{mapped_type} {barcode}"[:30]
+    description = fine_fee_type_mapping[fine_fee_type]
```
Comment (ghukill) [inline lambdas/bursar_transfer.py:120]: 👍

**Review by ghukill — APPROVED** [review]
Looks good to me!

**Review by ehanson8 — APPROVED** [review]

**Pull request merged**